src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -1100,6 +1100,82 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/{id}/authDomain:
+    get:
+      tags:
+        - snapshots
+        - repository
+      description: Retrieve the auth domain for the snapshot
+      operationId: getSnapshotAuthDomain
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      responses:
+        200:
+          description: Auth domain
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+        400:
+          description: Bad request - invalid id, badly formed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        403:
+          description: No permission to see auth domain
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        404:
+          description: Not found - snapshot id does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+    patch:
+      tags:
+        - snapshots
+        - repository
+      description: Patch the auth domain for the snapshot
+      operationId: patchSnapshotAuthDomain
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                type: string
+      responses:
+        200:
+          description: Auth domain
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PatchAuthDomainResponseModel'
+        400:
+          description: Bad request - invalid id, badly formed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        403:
+          description: No permission to see auth domain
```

Comment:
No worries! It took me some time to understand how it works too :) This is referring to the auth domain associated with a snapshot resource in Sam (I updated that description). It is separate from an auth domain attached to a workspace. If a snapshot doesn't have an auth domain, Sam will return an empty list. If a user tries to add an auth domain to the snapshot that includes a Terra user group that doesn't exist, Sam returns a 500 error. The 403 error actually shouldn't happen, because now all snapshot user roles include the "read_auth_domain" action in the [Sam config](https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L1274).

Reply:
Oh I think I understand now!  Your changes to documentation helped :)

So the caller is going to specify the collection of Sam user groups, and the flight will actually bundle them up into an auth domain (that's what is returned).

If the 403 shouldn't happen, can it be removed from the spec?

Reply:
Ah actually I got confused with the retrieve endpoint. If a snapshot reader tries to add an auth domain, they would get a 403 error. So this should say "No permission to add an auth domain to this snapshot"